crates/sui-network/src/discovery/server.rs

```diff
@@ -27,7 +31,32 @@ impl Discovery for Server {
             .our_info
             .clone()
             .ok_or_else(|| anemo::rpc::Status::internal("own_info has not been initialized yet"))?;
-        let known_peers = state.known_peers.values().cloned().collect();
+
+        // prefer returning peers that we are connected to as they are known-good
+        let mut known_peers = state
+            .connected_peers
+            .keys()
+            .filter_map(|peer_id| state.known_peers.get(peer_id))
+            .map(|info| (info.peer_id, info))
+            .take(MAX_PEERS_TO_SEND)
```

Comment:
should we also choose_multiple here so that we don't always send the same connected peers, if we are connected to >MAX_PEERS_TO_SEND peers?

Reply:
i think this is ok for now, since its a set of peers we know exist and most nodes likely have far fewer connections (our default is small iirc). That and it would be nice to avoid an extra allocation here

Reply:
Maybe add a warn! log if `known_peers.len() > MAX_PEERS_TO_SEND` just so we can remember to go back and address this? and/or a quick explanatory comment